tools/codegen/api/types.py

```diff
@@ -150,49 +150,36 @@ class CppSignature:
     # The schema this signature is derived from
     func: FunctionSchema
 
-    # Enough information about the C++ types to generate a full
-    # C++ type signature for this signature.  I'm not too sure
-    # if these are the right representations, so for now this
-    # is intended to be more abstract.
-    _argument_packs: Tuple[CppArgumentPack, ...]
-    _returns_type: str
+    # Is this a C++ signature for a method, i.e. Tensor::my_op(...)?
+    method: bool
+
+    # Is this a faithful C++ signature (i.e. following the JIT schema) or a convenience API
+    # (i.e. with a potential TensorOptions argument and out arguments in the front)
+    faithful: bool
 
     # Return the unpacked argument structure of this signature,
     # discarding information about which arguments are semantically
     # related to each other.
     def arguments(self) -> Sequence[CppArgument]:
-        return [sub_a for a in self._argument_packs for sub_a in a.explicit_arguments()]
+        return [sub_a for a in self.argument_packs() for sub_a in a.explicit_arguments()]
 
     # Return the packed argument structure of this signature.  This preserves
     # high-level structure of the arguments so you may find it easier to do
     # translations working with this representation.
     def argument_packs(self) -> Sequence[CppArgumentPack]:
-        return self._argument_packs
-
-    # Render the C++ declaration for this signature
-    def decl(self) -> str:
-        cpp_args_str = ', '.join(map(str, self.arguments()))
-        return f"{self._returns_type} {cpp.name(self.func)}({cpp_args_str})"
-
-    # Render the C++ definition for this signature, not including
-    # the body (with curly braces)
-    def defn(self, name: Optional[str] = None, *, prefix: str = "") -> str:
-        cpp_args_str = ', '.join(a.str_no_default() for a in self.arguments())
-        if name is None:
-            name = prefix + cpp.name(self.func)
-        return f"{self._returns_type} {name}({cpp_args_str})"
-
-    # NB: This constructor knows how to disambiguate defaults when
-    # faithful is True.  Ideally this would live as an external process
-    # see https://github.com/pytorch/pytorch/pull/45666
-    @staticmethod
-    def _from_grouped_arguments(
-        func: FunctionSchema,
-        arguments: Sequence[Union[Argument, TensorOptionsArguments, SelfArgument]],
-        *,
-        faithful: bool
-    ) -> 'CppSignature':
-        if faithful:
+        positional_and_kwargs: List[Union[Argument, SelfArgument, TensorOptionsArguments]] = \
```

Comment:
This segment is something like a copy paste of `group_arguments` but with some tweaks, right? I do think copy pasting is the right call here, but because I think at some point we should delete `group_arguments` entirely. Doesn't have to be this PR though.

Reply:
For this PR, it seems like a good idea to compare and contrast this to `group_arguments`

Reply:
yeah this is actually doing exactly what `group_arguments` is doing. Seems calling into `group_arguments` is a trivial simplification.